### PR TITLE
Split goroutine loop into 2 separate goroutines

### PR DESCRIPTION
**Pull request by henryprecheur**
This prevent one thing blocking the other

### CODE REVIEW

File: rest.go [hunk 1]
```diff
@@ -416,50 +416,56 @@ type Tunnel struct {
...
+			// FIXME old sauceconnect ignores error
+		} else if status != "running" {
+			//
+			// The tunnel is down, send its status back to the main loop.
+			//
+			t.ServerStatus <- status
+			close(t.ServerStatus)
+			return // We're done exit the loop
+		}
+		<-termTick
```
Comment (cstyan) [inline rest.go]: does `<-termTick` make the loop only loop every after `interval` amount of time has passed?

Comment (henryprecheur) [inline rest.go]: Yep, it will block until termTick is done.

Comment (cstyan) [inline rest.go]: hmm okay, just reads a bit awkwardly to me, what do you think about using `for range time.Tick(interval) {`?

Comment (henryprecheur) [inline rest.go]: Yep, that'd be more readable! I'll do it

**Comment by cstyan** [comment]
👍  lgtm

not sure if this is a big issue, but https://golang.org/pkg/time/#Tick says Tick can't be garbage collected

File: rest.go [hunk 2]
```diff
@@ -416,50 +417,50 @@ type Tunnel struct {
...
 			if err != nil {
 				// FIXME old sauceconnect ignores error
 			}
 		}
 	}
 }
 
+//
+// Goroutine that checks if the tunnel is still up and running, and sends a
+// heart beat to indicate the tunnel client is still up.
```
Comment (ldx) [inline rest.go]: I'd not call this "heartbeat" since we use the term for the POST request SC sends to REST.

**Comment by ldx** [comment]
:+1: 

Let's :ship: this! :)

**Pull request merged**